der/src/asn1/any.rs

```diff
@@ -226,19 +201,35 @@ pub struct Any {
     tag: Tag,
 
     /// Inner value encoded as bytes.
-    value: Vec<u8>,
+    value: ByteVec,
 }
 
 #[cfg(feature = "alloc")]
 impl Any {
     /// Create a new [`Any`] from the provided [`Tag`] and DER bytes.
-    pub fn new(tag: Tag, bytes: impl Into<Vec<u8>>) -> Result<Self> {
-        let value = bytes.into();
+    pub fn new(tag: Tag, bytes: &[u8]) -> Result<Self> {
+        let value = ByteVec::new(bytes)?;
```

Comment:
This was using an `impl Into<Vec<u8>>` constructor to allow a `Vec<u8>` to be provided as a parameter without making a needless copy.

I think `ByteVec::new` should do the same.

Reply:
It's only built from slices as far as I can tell, and with the move to a `Box<[u8]>` that would only optimize if there isn't excess capacity. Copy would happen otherwise.